src/app/InteractionModelEngine.cpp

```diff
@@ -1678,7 +1682,93 @@ void InteractionModelEngine::DispatchCommand(CommandHandlerImpl & apCommandObj,
 #endif // CHIP_CONFIG_USE_DATA_MODEL_INTERFACE
 }
 
-Protocols::InteractionModel::Status InteractionModelEngine::CommandExists(const ConcreteCommandPath & aCommandPath)
+Protocols::InteractionModel::Status InteractionModelEngine::ValidateCommandCanBeDispatched(const DataModel::InvokeRequest & request)
+{
+
+    Status status = CommandCheckExists(request.path);
+
+    if (status != Status::Success)
+    {
+        ChipLogDetail(DataManagement, "No command " ChipLogFormatMEI " in Cluster " ChipLogFormatMEI " on Endpoint 0x%x",
+                      ChipLogValueMEI(request.path.mCommandId), ChipLogValueMEI(request.path.mClusterId), request.path.mEndpointId);
+        return status;
+    }
+
+    status = CommandCheckACL(request);
+    VerifyOrReturnValue(status == Status::Success, status);
+
+    return CommandCheckFlags(request);
+}
+
+Protocols::InteractionModel::Status InteractionModelEngine::CommandCheckACL(const DataModel::InvokeRequest & aRequest)
+{
+    if (!aRequest.subjectDescriptor.has_value())
```

Comment:
Why is that subjectDescriptor field optional at all?  That seems like very strange API.

I guess for some read/write bits right now it doesn't exit.  But for commands it absolutely always must, and I think just aborting here if it does not is fine.

Reply:
SubjectDescriptor is currently part of the "common" part for all requests and it is optional since we try to model the ember "internal write" (i.e. we allow writes without needing ACL if internal flags are set). As a sideffect it is optional.

I think aborting is likely similar codes size as this and not crashing the entire app seems more resilient.

Reply:
OK.  It just makes it seem like there are "valid" case when it might be nullopt here, which there are not....